internal/mode/static/state/graph/route_common.go

```diff
@@ -529,6 +535,26 @@ func routeAllowedByListener(
 	return true
 }
 
+// isRouteKindAllowedByListener checks if the route kind is allowed by the listener.
+// If the listener does not specify allowed kinds, all kinds can attach to it.
```

Comment:
This should not be necessary, as when creating the listeners, we are adding supported kinds based on protocol if they are not specified.

Reply:
Yes agreed. But, this comment is more of a description for what the function does `when no kinds are added to the listener`

Reply:
Would that case of no kinds are added to the listener ever be reached? Since when creating the listener, we already perform that check

Reply:
The spec says:

> Kinds specifies the groups and kinds of Routes that are allowed to bind to this Gateway Listener. When unspecified or empty, the kinds of Routes selected are determined using the Listener protocol.

So, if the listener doesn't specify allowed kinds, then we determine the allowed kinds by looking at the Listener protocol. We don't alllow all kinds to attach.

Reply:
Oh okay. good to know. Let me modify this !